Week3/homework/step2-6.js

```diff
@@ -4,13 +4,18 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const new2d = arr.reduce(function(acc, curr) {
+    return Array.isArray(curr) ? acc.concat(curr) : acc.concat(curr);
+  }, []);
+  return new2d;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const new3d = arr.reduce(
+    (acc, curr) => (Array.isArray(curr) ? acc.concat(flattenArray3d(curr)) : acc.concat(curr)),
+    [],
+  );
+  return new3d;
```

Comment:
👍 
If you would like to take a step further, I would suggest that you have a look in Google to see how you could do the same thing using a concept called `recursion` 🙂

Reply:
I tried it but got lost in weird places while exploring. This will be on my list, recursion is scary :*(